src/util/nodeSelectors.ts

```diff
@@ -75,47 +75,63 @@ export function unwrapSelectionExtractor(
   };
 }
 
-export function selectWithLeadingDelimiter(
-  editor: TextEditor,
-  node: SyntaxNode
-): SelectionWithContext {
-  const leadingNonDelimiterToken = node.previousSibling?.previousSibling;
-
-  const leadingDelimiterRange =
-    leadingNonDelimiterToken != null
-      ? makeRangeFromPositions(
+export function selectWithLeadingDelimiter(...delimiters: string[]) {
+  return function (editor: TextEditor, node: SyntaxNode): SelectionWithContext {
+    const leadingDelimiterToken = node.previousSibling;
+    const leadingNonDelimiterToken = leadingDelimiterToken?.previousSibling;
+    let leadingDelimiterRange;
+    if (
+      leadingDelimiterToken &&
+      delimiters.includes(leadingDelimiterToken.type)
+    ) {
+      if (leadingNonDelimiterToken) {
+        leadingDelimiterRange = makeRangeFromPositions(
           leadingNonDelimiterToken.endPosition,
           node.startPosition
-        )
-      : null;
-
-  return {
-    ...simpleSelectionExtractor(editor, node),
-    context: {
-      leadingDelimiterRange,
-    },
+        );
+      } else {
+        leadingDelimiterRange = makeRangeFromPositions(
+          leadingDelimiterToken.startPosition,
+          node.startPosition
+        );
+      }
+    }
+    return {
+      ...simpleSelectionExtractor(editor, node),
+      context: {
+        leadingDelimiterRange,
+      },
+    };
   };
 }
 
-export function selectWithTrailingDelimiter(
-  editor: TextEditor,
-  node: SyntaxNode
-): SelectionWithContext {
-  const trailingNonDelimiterToken = node.nextSibling?.nextSibling;
-
-  const trailingDelimiterRange =
-    trailingNonDelimiterToken != null
-      ? makeRangeFromPositions(
+export function selectWithTrailingDelimiter(...delimiters: string[]) {
+  return function (editor: TextEditor, node: SyntaxNode): SelectionWithContext {
+    const trailingDelimiterToken = node.nextSibling;
+    const trailingNonDelimiterToken = trailingDelimiterToken?.nextSibling;
+    let trailingDelimiterRange;
+    if (
+      trailingDelimiterToken &&
+      delimiters.includes(trailingDelimiterToken.type)
+    ) {
+      if (trailingNonDelimiterToken) {
+        trailingDelimiterRange = makeRangeFromPositions(
           node.endPosition,
           trailingNonDelimiterToken.startPosition
-        )
-      : null;
-
-  return {
-    ...simpleSelectionExtractor(editor, node),
-    context: {
-      trailingDelimiterRange,
-    },
+        );
+      } else {
+        trailingDelimiterRange = makeRangeFromPositions(
```

Comment:
What if we walked up parents until we find one with a previous sibling and then ask for end of previous sibling there? Not sure it's worth the effort but might make white space a bit cleaner

Reply:
We can't try that

Reply:
Did you mean to say "we _can_ try that?"

Reply:
I don't think we have to. So far no language has need of this with the current implementation.